code/go/internal/validator/common_spec.go

```diff
@@ -5,14 +5,51 @@
 package validator
 
 import (
+	"reflect"
+
 	"github.com/creasty/defaults"
 	"github.com/pkg/errors"
+
+	"github.com/elastic/package-spec/code/go/internal/spectypes"
 )
 
 type commonSpec struct {
 	AdditionalContents bool             `yaml:"additionalContents"`
 	Contents           []folderItemSpec `yaml:"contents"`
 	DevelopmentFolder  bool             `yaml:"developmentFolder"`
+
+	Limits commonSpecLimits `yaml:",inline"`
+}
+
+type commonSpecLimits struct {
+	// Limit to the total number of elements in a directory.
+	TotalContentsLimit int `yaml:"totalContentsLimit"`
+
+	// Limit to the total size of files in a directory.
+	TotalSizeLimit spectypes.FileSize `yaml:"totalSizeLimit"`
+
+	// Limit to individual files.
+	SizeLimit spectypes.FileSize `yaml:"sizeLimit"`
+
+	// Limit to individual configuration files (yaml files).
+	ConfigurationSizeLimit spectypes.FileSize `yaml:"configurationSizeLimit"`
+
+	// Limit to files referenced as relative paths (images).
+	RelativePathSizeLimit spectypes.FileSize `yaml:"relativePathSizeLimit"`
+
+	// Maximum number of fields per data stream, can only be set at the root level spec.
+	FieldsPerDataStreamLimit int `yaml:"fieldsPerDataStreamLimit"`
+}
+
+func (l *commonSpecLimits) update(o commonSpecLimits) {
```

Comment:
In general, I'm hesitant to use reflection unless we can't achieve something any other way round. I'm wondering if we can go with a standard copy operation between fields.

Reply:
I agree, but for this case I didn't find any other way around:
* Preinitializing the structs with default values and a standard copy is not possible, because these objects are created when unmarshaling the objects. And these "default" values are obtained in the same unmarshal operation.
* I could do some kind of unmarshaling in two passes, one to read the limits on the parent and preinitialize the children contents, and another one to unmarshal the children, bit this can be worse than using reflection 
* I could do the good old set field by field, but I consider this error prone, we may forget of adding a field here after adding it to the struct.

Reply:
> I could do some kind of unmarshaling in two passes, one to read the limits on the parent and preinitialize the children contents, and another one to unmarshal the children, bit this can be worse than using reflection

Yes, I used to follow this option when interacting with complex structures, ignoring the performance issues. Regarding reflection, I'm not sure if it still works when we add slices or pointers. 

Let's stick to reflection, but maybe make sure if it's supported if we decide to build a WebAssembly out of the package-spec. It might be a blocker then.

Reply:
> Yes, I used to follow this option when interacting with complex structures, ignoring the performance issues. Regarding reflection, I'm not sure if it still works when we add slices or pointers.

Yep, with more complex types this may not work, but this object is intended to store limits, that are most likely going to be numeric values.

> Let's stick to reflection, but maybe make sure if it's supported if we decide to build a WebAssembly out of the package-spec. It might be a blocker then.

I wouldn't expect compatibility issues because of using reflection, this is also used by core packages such as `encode/json`.

(In any case, I have tried, and tests pass with WebAssembly - `GOOS=js GOARCH=wasm go test -exec="$(go env GOROOT)/misc/wasm/go_js_wasm_exec" ./...`)